api/server_helpers_test.go

```diff
@@ -24,6 +29,102 @@ import (
 	"github.com/NebulousLabs/Sia/types"
 )
 
+// A Server is essentially a collection of modules and an API server to talk
+// to them all.
+type Server struct {
+	api *API
+
+	apiServer         *http.Server
+	listener          net.Listener
+	requiredUserAgent string
+
+	// wg is used to block Close() from returning until Serve() has finished. A
+	// WaitGroup is used instead of a chan struct{} so that Close() can be called
+	// without necessarily calling Serve() first.
+	wg sync.WaitGroup
+}
+
+// NewServer creates a new API server from the provided modules. The API will
+// require authentication using HTTP basic auth if the supplied password is not
+// the empty string. Usernames are ignored for authentication. This type of
+// authentication sends passwords in plaintext and should therefore only be
+// used if the APIaddr is localhost.
+func NewServer(APIaddr string, requiredUserAgent string, requiredPassword string, cs modules.ConsensusSet, e modules.Explorer, g modules.Gateway, h modules.Host, m modules.Miner, r modules.Renter, tp modules.TransactionPool, w modules.Wallet) (*Server, error) {
+	l, err := net.Listen("tcp", APIaddr)
+	if err != nil {
+		return nil, err
+	}
+
+	a := New(requiredUserAgent, requiredPassword, cs, e, g, h, m, r, tp, w)
+	serv := &http.Server{
+		Handler: a,
+	}
```

Comment:
inline this in the `Server` struct for better readability (no `serv`/`srv` distinction)